keanu-project/src/main/java/io/improbable/keanu/algorithms/graphtraversal/VertexValuePropagation.java

```diff
@@ -125,4 +125,89 @@ private static void cascadeUpdate(Deque<Vertex> stack, Map<Long, Long> explored)
 
         }
     }
+
+    public static void eval(Vertex... vertices) {
+        eval(Arrays.asList(vertices));
+    }
+
+    public static void eval(Collection<? extends Vertex> vertices) {
```

Comment:
`eval` and `lazyEval` are so similar that I think it will actually be simpler to understand if we put them under one function and use a conditional for the `parentsThatAreNotCalculated` like:

```
 public static void eval(Collection<? extends Vertex> vertices, boolean checkValueOrVisited) {
        Deque<Vertex<?>> stack = new ArrayDeque<>();
        for (Vertex v : vertices) {
            stack.push(v);
        }

        Set<Vertex<?>> hasCalculated = new HashSet<>();

        while (!stack.isEmpty()) {

            Vertex<?> head = stack.peek();
            Set<Vertex<?>> parentsThatAreNotYetCalculated = checkValueOrVisited ?
                parentsThatAreNotCalculated(hasCalculated, head.getParents()) : parentsThatAreNotCalculated(head.getParents());

            if (head.isProbabilistic() || parentsThatAreNotYetCalculated.isEmpty()) {

                Vertex<?> top = stack.pop();
                top.updateValue();
                hasCalculated.add(top);

            } else {

                for (Vertex<?> vertex : parentsThatAreNotYetCalculated) {
                    stack.push(vertex);
                }

            }

        }
    }
```